packages/integration-tests/tests/connectors.test.ts

```diff
@@ -34,8 +82,45 @@ test('connector flow', async () => {
   expect(facebookConnector.enabled).toBeTruthy();
   expect(facebookConnector.config).toEqual(facebookConnectorConfig);
 
+  /*
+   * Set up an SMS connector
+   */
+  const updatedAliyunSmsConnector = await updateConnectorConfig(
+    aliyunSmsConnectorId,
+    aliyunSmsConnectorConfig
+  );
+  expect(updatedAliyunSmsConnector.config).toEqual(aliyunSmsConnectorConfig);
+  const enabledAliyunSmsConnector = await enableConnector(aliyunSmsConnectorId);
+  expect(enabledAliyunSmsConnector.enabled).toBeTruthy();
+
+  /*
+   * Change to another SMS connector
+   */
+  const updatedTwilioSmsConnector = await updateConnectorConfig(
+    twilioSmsConnectorId,
+    twilioSmsConnectorConfig
+  );
+  expect(updatedTwilioSmsConnector.config).toEqual(twilioSmsConnectorConfig);
+  const enabledTwilioSmsConnector = await enableConnector(twilioSmsConnectorId);
+  expect(enabledTwilioSmsConnector.enabled).toBeTruthy();
+
+  // There should be exactly one enabled SMS connector after changing to another SMS connector.
+  const connectorsAfterChangingSmsConnector = await listConnectors();
+  expect(
+    connectorsAfterChangingSmsConnector.find(
+      (connector) => connector.id === twilioSmsConnectorId && connector.enabled
+    )
+  ).toBeTruthy();
+  expect(
+    connectorsAfterChangingSmsConnector.find(
+      (connector) =>
+        connector.type === ConnectorType.SMS &&
+        connector.id !== twilioSmsConnectorId &&
+        connector.enabled
+    )
+  ).toBeFalsy();
```

Comment:
nit: could simply filter conenctorType = SMS && enabled. length === 1. and id === twilioSmsConnectorId

Reply:
Good call. Refactored

```ts
  const enabledSmsConnectors = connectorsAfterChangingSmsConnector.filter(
    (connector) => connector.type === ConnectorType.SMS && connector.enabled
  );
  expect(enabledSmsConnectors.length).toEqual(1);
  expect(enabledSmsConnectors[0]?.id).toEqual(twilioSmsConnectorId);

```